procedural/processor/src/main/java/gov/nasa/ammos/aerie/procedural/processor/SchedulingProcedureProcessor.java

```diff
@@ -0,0 +1,284 @@
+
+import javax.annotation.processing.Completion;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.Messager;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.AnnotationValue;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.PackageElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import javax.tools.Diagnostic;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Repeatable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+public final class SchedulingProcedureProcessor implements Processor {
+  // Effectively final, late-initialized
+  private Messager messager = null;
+  private Filer filer = null;
+  private Elements elementUtils = null;
+  private Types typeUtils = null;
+
+  @Override
+  public Set<String> getSupportedOptions() {
+    return Set.of();
+  }
+
+  /** Elements marked by these annotations will be treated as processing roots. */
+  @Override
+  public Set<String> getSupportedAnnotationTypes() {
+    return Set.of(SchedulingProcedure.class.getCanonicalName(), WithMappers.class.getCanonicalName());
+  }
+
+  @Override
+  public SourceVersion getSupportedSourceVersion() {
+    return SourceVersion.latest();
+  }
+
+  @Override
+  public void init(final ProcessingEnvironment processingEnv) {
+    this.messager = processingEnv.getMessager();
+    this.filer = processingEnv.getFiler();
+    this.elementUtils = processingEnv.getElementUtils();
+    this.typeUtils = processingEnv.getTypeUtils();
+  }
+
+  @Override
+  public boolean process(final Set<? extends TypeElement> annotations, final RoundEnvironment roundEnv) {
+    final var mapperClassElements = new ArrayList<TypeElement>();
+    PackageElement packageElement = null;
+    for (final var packageElement$ : roundEnv.getElementsAnnotatedWith(WithMappers.class)) {
+      if (packageElement != null) throw new RuntimeException("Multiple packages annotated with WithMappers");
+      if (packageElement$.getKind() != ElementKind.PACKAGE) throw new RuntimeException("Only packages can be annotated with WithMappers");
+      packageElement = (PackageElement) packageElement$;
+    }
+    if (packageElement == null) return false; //throw new RuntimeException("Need to annotate a package-info class with WithMappers");
```

Comment:
Flagging this commented out `throw`

Reply:
@mattdailis Would you rather keep this as an exception? I.e. uncomment instead of return false?

Reply:
Matt is fine removing this. I've made the change